src/profile-logic/process-profile.js

```diff
@@ -470,7 +472,32 @@ function _processStackTable(geckoStackTable: GeckoStackStruct): StackTable {
  */
 function _processMarkers(geckoMarkers: GeckoMarkerStruct): MarkersTable {
   return {
-    data: geckoMarkers.data,
+    data: geckoMarkers.data.map(function(m: Object): MarkerPayload {
+      if (m) {
+        switch (m.type) {
+          case 'GCSlice':
+            if (m.timings && m.timings.times) {
+              m.timings.phase_times = convertPhaseTimes(m.timings.times);
+              delete m.timings.times;
+            } else {
+              m.timings.phase_times = [];
+            }
+            break;
+          case 'GCMajor':
+            if (m.timings.status === 'completed') {
+              const timings = m.timings;
+              timings.phase_times = convertPhaseTimes(timings.phase_times);
+              timings.mmu_20ms /= 100;
+              timings.mmu_50ms /= 100;
+              timings.allocated *= 1024 * 1024;
+            }
+            break;
+          default:
+            break;
+        }
+      }
+      return m;
+    }, geckoMarkers),
```

Comment:
this last argument is to define the "this", but you don't use `this` at all ?

Reply:
Also, I don't think this is the right location for this. Did you want to put this in `processed-profile-versioning.js` ?

Reply:
No, I'm pretty sure I want it in process-profile.js.  It's for making the processed profile more friendly. Where as the gecko profile has to be telemetry-compatible.  If you disagree please tell me.I'm not aware of the "this" thing with higher-order coding in JS.  I'm happy for you to correct my bad JS.

Reply:
about the "this" thing, this is especially useful when you want to use an instance's method in `map` or other iteration functions, for example like this:
```javascript
class Something {
  doSomething() {
    array.map(this.method, this);
  }
}
```
Without the final `this`, the `this` in `method` would be `null`. `This` handling in javascript is weird. This has been a real issue before we had the arrow functions (normal functions change the `this` while arrow functions keep the `this` from outside their scope).

Reply:
Ah, I see. Thanks.